src/sysupdate/sysupdate.c

```diff
@@ -976,12 +1056,28 @@ static int verb_check_new(int argc, char **argv, void *userdata) {
         if (r < 0)
                 return r;
 
-        if (!context->candidate) {
-                log_debug("No candidate found.");
-                return EXIT_FAILURE;
+        if (FLAGS_SET(arg_json_format_flags, SD_JSON_FORMAT_OFF)) {
+                if (!context->candidate) {
+                        log_debug("No candidate found.");
+                        return EXIT_FAILURE;
+                }
+
+                puts(context->candidate->version);
+        } else {
+                _cleanup_(sd_json_variant_unrefp) sd_json_variant *json = NULL;
+
+                if (context->candidate)
+                        r = sd_json_variant_new_string(&json, context->candidate->version);
+                else
+                        r = sd_json_variant_new_null(&json);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to create JSON: %m");
+
+                r = sd_json_variant_dump(json, arg_json_format_flags, stdout, NULL);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to print JSON: %m");
         }
```

Comment:
things like JSON-I say that the top-level json variant should always be object/array and not a specific type. also, this here is not really extensible, and we probably should provide some extensibility here.

hence, please wrap this in a trivial object with just one field

Reply:
I added a review comment about this on the other PR; resolving this to avoid duplicate